File: XPile 3-17/Programs/test_nested_fn.py
from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def outer():
    def add_ten(x):
        return (x + 10)
    
    def twice(x):
        return (x * 2)
    
    a = add_ten(5)
    b = twice(a)
    print(a)
    print(b)
    return b

def main():
    def greet():
        return 'hello from nested'
    
    result = outer()
    print(result)
    msg = greet()
    print(msg)
    return 0


if __name__ == "__main__":
    main()

File: XPile 3-17/Programs/test_nested_fn.js

function outer() {
    function add_ten(x) {
        return (x + 10);
    }
    
    function twice(x) {
        return (x * 2);
    }
    
    let a = add_ten(5);
    let b = twice(a);
    console.log(a);
    console.log(b);
    return b;
}

function main() {
    function greet() {
        return "hello from nested";
    }
    
    let result = outer();
    console.log(result);
    let msg = greet();
    console.log(msg);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();
